File: 228-summary-ranges/summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        result = []
        i = 0

        while i < len(nums):
            start = nums[i]
            while i < len(nums) - 1 and nums[i] + 1 == nums[i + 1]:
                i+=1
            
            if start != nums[i]:
                result.append(f"{start}->{nums[i]}")
            else:
                result.append(f"{nums[i]}")
            i += 1

        return result

File: 228-summary-ranges/summary-ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
    const result = [];
  let i = 0;

  // Scan each range starting point once
  while (i < nums.length) {
    const start = nums[i];

    // Extend while numbers stay consecutive
    while (i + 1 < nums.length && nums[i + 1] === nums[i] + 1) {
      i++;
    }

    const end = nums[i];

    // Add single value or full range based on boundaries
    if (start === end) {
      result.push(String(start));
    } else {
      result.push(`${start}->${end}`);
    }

    i++;
  }

  return result;
};